src/main/java/com/github/manolo8/darkbot/gui/MapDrawer.java

```diff
@@ -124,6 +152,20 @@ private void drawActionButton() {
         }
     }
 
+    @EventHandler
+    public void onMapChange(StarSystemAPI.MapChangeEvent e) {
+        if (e.getNext().getId() <= 0) {
+            backgroundImage = null;
+            return;
+        }
+
+        if (main.config.BOT_SETTINGS.CUSTOM_BACKGROUND.ENABLED
+                && main.config.BOT_SETTINGS.CUSTOM_BACKGROUND.USE_GAME_BACKGROUND) {
+            new BackgroundImgDownloaderTask(this, e.getNext().getId())
+                    .execute();
+        }
+    }
```

Comment:
I think this should just be resolved on rendering time, otherwise it will only work after jumping map when you select a map.

You can just store a "int lastMapId", and update it when the image is changed, and avoid the Listener in general